src/nvim/fileio.c

```diff
@@ -2879,8 +2883,21 @@ buf_write (
           }
 #endif
 
+#ifdef HAVE_READLINK
+          // In the case where the file is a symlink, we want to copy the link
+          // target instead of the link itself.
```

Comment:
would be good to mention the related ticket.

```suggestion
          // target instead of the link itself. #10101
```

Reply:
Will do